cmd/uplink/cmd/setup.go

```diff
@@ -247,3 +271,34 @@ func ApplyDefaultHostAndPortToAddr(address, defaultAddress string) (string, erro
 	// address is host:
 	return net.JoinHostPort(addressParts[0], defaultPort), nil
 }
+
+// saveEncryptionKey generates a Storj key from the inputKey and save it into a
+// new file created in filepath.
+func saveEncryptionKey(inputKey []byte, filepath string) error {
+	switch {
+	case len(inputKey) == 0:
+		return Error.New("inputKey is empty")
+	case filepath == "":
+		return Error.New("filepath is empty")
+	}
+
+	file, err := os.OpenFile(filepath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return Error.New("directory path doesn't exist. %+v", err)
+		}
+
+		if os.IsExist(err) {
+			return Error.New("file key already exists. %+v", err)
+		}
+
+		return Error.Wrap(err)
+	}
+
+	defer func() {
+		err = Error.Wrap(errs.Combine(err, file.Close()))
+	}()
+
+	_, err = file.Write(inputKey)
```

Comment:
Could you simplify the implementation of this method by taking advantage of https://golang.org/pkg/io/ioutil/#WriteFile?

Reply:
I cannot if what we want is not to override an existing file, if it doesn't matter if the file exists and can be overridden then let me know and I swap over it.

Reply:
OK. Then leave it like this.